sound/soc/intel/boards/broadwell.c

```diff
@@ -150,17 +150,44 @@ static int broadwell_rtd_init(struct snd_soc_pcm_runtime *rtd)
 	return 0;
 }
 
+static struct snd_soc_dai_link_component dummy_codec_component[] = {
+	{
+		.name = "snd-soc-dummy",
+		.dai_name = "snd-soc-dummy-dai"
+	},
+};
+
+static struct snd_soc_dai_link_component rt286_component[] = {
+	{
+		.name = "i2c-INT343A:00",
+		.dai_name = "rt286-aif1",
+	}
+};
+
+static struct snd_soc_dai_link_component platform_component[] = {
+	{
+		.name = "haswell-pcm-audio"
+	}
+};
+
+static struct snd_soc_dai_link_component platform_dummy_component[] = {
+	{
+		.name = "snd-soc-dummy"
+	}
+};
```

Comment:
We're going to use identical dummy_codec_component[] and platform_dummy_component[] in multiple files, maybe define them once in a common file and use them from all? Many (distribution) kernel configurations build support for all platforms, but only one is used at run-time, so, even if those structs change at run-time, that shouldn't be a problem.

Reply:
Yes, I've thought about this but at the same time if we add stuff to a common file then we'll get warnings that a structure is defined but not used. Using a global across multiple modules is also not great. I don't have a good solution so far.

Reply:
Could we add it to a new created include/sound/soc-utils.h and include the soc-utils.h in the machine driver which need this?

Reply:
I'll just try to remove the platform_name first, and we'll see if we can remove the dummy_component for the codecs later.